File: src/Einstein.WebUI/ClientApp/components/work/employees/models/Employee.js

export class Employee {
	constructor({
		seqno = 0,

		status = {
			valid: true,
			added: false,
			modified: false,
			removed: false,
			loaded: false
		},

		id = -1,
		deleted = false,

		employeeNumber = -1,

		firstname = '',
		lastname = '',
		email = '',

		employedSince = null,


		availabilityPerWeek = 100,

		functionLevel = 'junior',

		functionTitle = 'Software Developer',

		managerId = -1

	} = {}) {

		this.id = id;
		this.deleted = deleted;

		this.seqno = seqno;
		this.status = status;

		this.employeeNumber = employeeNumber,

		this.firstname = firstname,
		this.lastname = lastname,
		this.email = email,

		this.employedSince = employedSince,

		this.availabilityPerWeek = availabilityPerWeek,

		this.functionLevel = functionLevel,

		this.functionTitle = functionTitle,

		this.managerId = managerId
	}
}

export function createEmployee(data) {
	return Object.freeze(new Employee(data));
}

File: src/Bloemert.Data.Entity.Work/Entity/Employee.cs
using Bloemert.Data.Core;
using Newtonsoft.Json;
using System;

namespace Bloemert.Data.Entity.Work.Entity
{
	public class Employee : BaseEntity
	{
		public virtual string Firstname { get; set; }
		public virtual string Lastname { get; set; }
		public virtual string Email { get; set; }


		public virtual DateTime? EmployedSince { get; set; }

		public virtual int EmployeeNumber { get; set; }

		public virtual int AvailabilityPerWeek { get; set; }

		public virtual string FunctionLevel { get; set; }

		public virtual string FunctionTitle { get; set; }

		public virtual Employee Manager { get; set; }
	}
}
